addon/components/o-s-s/search-field.ts

```diff
@@ -0,0 +1,33 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import Component from '@glimmer/component';
+import type IntlService from 'ember-intl/services/intl';
+
+interface OSSSearchFieldArgs {
+  value: string;
+  placeholder?: string;
+  // eslint-disable-next-line no-unused-vars
+  onChange: (value: string) => void;
+}
+
+export default class OSSSearchFieldComponent extends Component<OSSSearchFieldArgs> {
+  @service declare intl: IntlService;
+  @tracked value = this.args.value;
```

Comment:
instead of copying `this.args.value` into a local `this.value` property, it's better to always use `this.args.value` (`@value` in the template) so that the code using this component has the full ownership of the `value`.

Having a local `value` means that there is a risk of having an out-of-sync input.
For example, let's say the `onChange` function passed to this component has the following code:

```
onChange(valueFromSearchField: string) {
  this.valueFromSearchField = `${valueFromSearchField} suffix`
}
```

In that case if you search for "hello" you'll have a weird situation where:
- the parent stores `hello suffix`
- but the input still displays `hello`